pkg/api/dashboard.go

```diff
@@ -49,6 +54,33 @@ func dashboardGuardianResponse(err error) response.Response {
 func (hs *HTTPServer) GetDashboard(c *models.ReqContext) response.Response {
 	slug := c.Params(":slug")
 	uid := c.Params(":uid")
+	//Apply Patch
+	validSharedUid := false
+	if !c.IsSignedIn && len(c.QueryStrings("shareduid")) > 0 {
+		text := strings.Join(c.QueryStrings("shareduid"), "")
+		key := []byte("p0w3r3dByGr4f4n4")
```

Comment:
super secret key?

Reply:
> > This feature is a real sharing link, for any user, including anonymous users.
> 
> Can you describe how this works? what this PR does? How would query access be handled? How does the security work in this PR? For a big feature like this it could be good to start with a design doc with the problem your trying to solve and your proposed solution.

Sorry about the low documenting. This is my first PR to a big project. Basically, this public link is mount with a hash based in the uid and slug from dashboard, using the "super secret key", as you mentioned. The idea is share the dashboard as a viewer to any user.  

Here is a example:
https://opmon.opservices.com/grafana/d/uIwDF9Tiz/opmon-tactical-view?orgId=1&shareduid=UTabSWDZ8EIymOD2hUTxHIpBjJ1Q4UOqwOYsJZcvqONiynXqVBWU6TyaDCAY

The query access is the same. There is just a validation of the hash. I'm sure there is issues to improve this proposal feature, and please let me know.